### PR TITLE
Add ConsensusQueue to handle validation test suite

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

Adds back `ConsensusQueue` to the handle validation tests. The test had a few issues preventing it from working with ConsensusQueue. The direct cause for the hang that the original bug reported is that container1 was disposed before attempting to read the handle, and ConsensusQueue requires ops to round-trip before its data can be read.

While working on this, I also refactored the affected test and suite a fair bit for clarity, and tweaked what the test case was doing. I split it into a couple different cases based on what the handle in the DDS was referencing, as [previous discussion](https://github.com/microsoft/FluidFramework/pull/20995#discussion_r1591574035) on the topic mentioned the problematic case was something not being explicitly tested.

[AB#8058](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/8058)

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts [hunk 1]
```diff
@@ -69,10 +76,59 @@ function treeSetup(dds: ITree) {
 	return view;
 }
 
-async function setup(
-	getTestObjectProvider: (options?: ITestObjectProviderOptions) => ITestObjectProvider,
-	apis: CompatApis,
-) {
+interface aDDSType {
```
Comment (Abe27342) [inline packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts:79]: i was considering renaming these to `HandleStorage` and `HandleStorageFactory`, as I think it'd make it a bit clearer. If reviewer is ok with it, I can do a bunch of renames in a separate commit to make this whole file more consistent in naming (didn't want to make review too challenging, it's already a bit annoying as I moved some code around)

File: packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts [hunk 2]
```diff
@@ -115,287 +406,165 @@ async function setup(
...
+	/**
+	 * Loads a container and waits for its sequence number to catch up to some existing container (at the time the function begins).
+	 */
+	async function loadAndCatchUpContainer(
+		provider: ITestObjectProvider,
+		existingContainer: IContainer,
+	): Promise<IContainer> {
+		const seq = existingContainer.deltaManager.lastSequenceNumber;
+		const newContainer = await provider.loadTestContainer(testContainerConfig);
+		if (newContainer.deltaManager.lastSequenceNumber < seq) {
```
Comment (anthony-murphy) [inline packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts:435]: can we just reuse waitContainerToCatchUp

File: packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts [hunk 3]
```diff
@@ -489,17 +655,14 @@ describeCompat("handle validation", "NoCompat", (getTestObjectProvider, apis) =>
 		});
 	}
 
-	for (const detachedDds1Utils of ddsTypes) {
-		for (const detachedDds2Utils of ddsTypes) {
-			for (const attachedDdsUtils of ddsTypes) {
+	for (const detachedDds1Utils of handleStorageFactories) {
```
Comment (anthony-murphy) [inline packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts:658]: can we switch this to pairwise as well.

**Review by anthony-murphy — APPROVED** [review]

File: packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts (hunk 1 above)
Comment (anthony-murphy) [reply packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts:79]: maybe just do it in a follow up to keep this change reasonable. should be quick to review and approve a simple rename.

File: packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts (hunk 2 above)
Comment (Abe27342) [reply packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts:435]: yeah, seems like that should suffice here.

**Pull request merged**